presto-iceberg/src/main/java/com/facebook/presto/iceberg/util/StatisticsUtil.java

```diff
@@ -57,29 +68,49 @@ public static TableStatistics mergeHiveStatistics(TableStatistics icebergStatist
                     .setDistinctValuesCount(icebergColumnStats.getDistinctValuesCount())
                     .setRange(icebergColumnStats.getRange());
             if (hiveColumnStats != null) {
-                if (mergeStrategy.equals(USE_NDV) || mergeStrategy.equals(USE_NULLS_FRACTION_AND_NDV)) {
+                // NDVs
+                if (mergeFlags.contains(NUMBER_OF_DISTINCT_VALUES)) {
                     hiveColumnStats.getDistinctValuesCount().ifPresent(ndvs -> mergedStats.setDistinctValuesCount(Estimate.of(ndvs)));
                 }
-                if (mergeStrategy.equals(USE_NULLS_FRACTIONS) || mergeStrategy.equals(USE_NULLS_FRACTION_AND_NDV)) {
-                    hiveColumnStats.getNullsCount().ifPresent(nullCount -> {
-                        Estimate nullsFraction;
-                        if (!hiveStatistics.getBasicStatistics().getRowCount().isPresent()) {
-                            if (icebergStatistics.getRowCount().isUnknown()) {
-                                nullsFraction = Estimate.unknown();
-                            }
-                            else {
-                                nullsFraction = Estimate.of((double) nullCount / icebergStatistics.getRowCount().getValue());
-                            }
-                        }
-                        else {
-                            nullsFraction = Estimate.of((double) nullCount / hiveStatistics.getBasicStatistics().getRowCount().getAsLong());
-                        }
-                        mergedStats.setNullsFraction(nullsFraction);
-                    });
+                // data size
+                if (mergeFlags.contains(ColumnStatisticType.TOTAL_SIZE_IN_BYTES)) {
+                    hiveColumnStats.getTotalSizeInBytes().ifPresent(size -> mergedStats.setDataSize(Estimate.of(size)));
                 }
             }
             statsBuilder.setColumnStatistics(columnHandle, mergedStats.build());
         });
-        return statsBuilder.build();
+        return calculateAndSetTableSize(statsBuilder).build();
+    }
+
+    public static EnumSet<ColumnStatisticType> decodeMergeFlags(String input)
+    {
+        return Optional.of(Arrays.stream((input).trim().split(","))
+                        .filter(value -> !value.isEmpty())
+                        .map(ColumnStatisticType::valueOf)
```

Comment:
Let's also add a check here that the mapped ColumnStatisticType is one of {NUMBER_OF_DISTINCT_VALUES, TOTAL_SIZE_IN_BYTES} since that is all we support at the moment ?